pkg/frontend/querymiddleware/codec.go

```diff
@@ -149,6 +149,16 @@ type prometheusCodec struct {
 	requestProtobufFormat bool
 }
 
+type format interface {
+	EncodeResponse(resp *PrometheusResponse) ([]byte, error)
+	DecodeResponse([]byte) (PrometheusResponse, error)
```

Comment:
Any reason that this method returns `PrometheusResponse` vs `*PrometheusResponse` here? I don't know that it matters, just seems a little odd since we immediately grab a pointer to it in the `prometheusCodec.DecodeResponse` below.

Reply:
Good point - I think that was an artifact of how things were set up pre-refactoring and I didn't think to change it.

Fixed in ea59bdc.